pytorch_lightning/profiler/pytorch.py

```diff
@@ -448,68 +613,242 @@ def on_trace_ready(profiler):
             if _TORCH_GREATER_EQUAL_1_9:
                 self.profiler.add_metadata("Framework", "pytorch-lightning")
 
-    def summary(self) -> str:
-        if not self._profiler_kwargs.get("enabled", True) or self._emit_nvtx:
-            return ""
+    def _delete_profiler(self) -> None:
+        if self.profiler is not None:
+            self.profiler.__exit__(None, None, None)
+            self._cache_functions_events()
+            self.profiler = None
 
-        self._delete_profilers()
+        if self._schedule is not None:
+            self._schedule.reset()
 
-        if not self.function_events:
-            return ""
 
-        if self._export_to_chrome and not _KINETO_AVAILABLE:
-            filename = f"{self.local_rank}_trace.json"
-            path_to_trace = filename if self.dirpath is None else os.path.join(self.dirpath, filename)
-            self.function_events.export_chrome_trace(path_to_trace)
+class KinetoProfilerState(IntEnum):
```

Comment:
Could we use the `ProfilerAction` enum defined in `torch.profiler`?

If not, does this class need to be public?

Reply:
will make private